jsonapi/data_structs.go

```diff
@@ -48,6 +48,15 @@ func (c *DataContainer) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.DataObject)
 }
 
+// CustomLink represents a custom link for return in the document.
+type CustomLink struct {
```

Comment:
hm, a link in the `data` part can either be an object with `href` and `meta` or just a string, right? How about we make this data structure with an additional field that would be the string and then we can also change the `map[string]interface{}` to directly point to this struct. A custom marshaler/unmarshaler would be needed then